test/parallel/test-fs-watch-file-nonexistent.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+var fs = require('fs');
+
+var filename = '/path/to/file/that/does/not/exist';
+
+fs.watchFile(filename, function() {
+  throw new Error('callback should not be called for non-existent files');
```

Comment:
We can leave this in and mark the current functionality as correct. I marked the issue as `confirmed-bug` because folks in joyent/node did and ported the PR - but I don't have a strong opinion on whether or not the callback should be called.